slatedb/src/merge_iterator.rs

```diff
@@ -442,12 +449,58 @@ mod tests {
         assert_iterator(
             &mut merge_iter,
             vec![
-                RowEntry::new_value(b"key1", b"value1", 1), // first occurrence
                 RowEntry::new_value(b"key1", b"value1_updated", 3), // second occurrence
+                RowEntry::new_value(b"key1", b"value1", 1),         // first occurrence
                 RowEntry::new_value(b"key2", b"value2", 2),
                 RowEntry::new_value(b"key3", b"value3", 4),
             ],
         )
         .await;
     }
```

Comment:
this is the test that had the wrong behavior. you can see that before it actually returned `key1 | seqnum1` before `key1 | seqnum3`